### PR TITLE
Add Rate Limit and Captcha verification fo Aptos

### PR DESCRIPTION
**Pull request by 0xPrimata**

### CODE REVIEW

File: api/rate-limit.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+function getXForwardedFor(req: ExtendedIncomingMessage): string | undefined {
+  return req.headers['x-forwarded-for'];
+}
+
+function ips(req: any) {
+return getXForwardedFor(req)?.split(/\s*,\s*/);
+}
+
+export default async function handler(request: any, response: any) {
+  // You could alternatively limit based on user ID or similar
```
Comment (l-monninger) [inline api/rate-limit.ts:27]: You should place your captcha validation here.

**Review by l-monninger — CHANGES_REQUESTED** [review]

**Comment by 0xPrimata** [comment]
> 2. I would adjust the form so that the captcha doesn't immediately uncheck and require re-auth.

2. didn't get what you meant, it only unchecks if it passes rate limit or if there is an issue with captcha

**Comment by l-monninger** [comment]
> 2\. didn't get what you meant, it only unchecks if it passes rate limit or if there is an issue with captcha

For me, it unchecks when the request succeeds too.

**Comment by 0xPrimata** [comment]
> For me, it unchecks when the request succeeds too.

got it, done! @l-monninger

File: src/pages/LandingPage/Index.tsx [hunk 2]
```diff
@@ -187,10 +191,9 @@ export default function LandingPage() {
 
   const aptosFaucetRequest = async (address: string, token: string) => {
```
Comment (l-monninger) [inline src/pages/LandingPage/Index.tsx:192]: It doesn't look like you are actually using these for anything. Please remove.

File: src/pages/LandingPage/Index.tsx [hunk 3]
```diff
@@ -17,6 +17,10 @@ import evmTokensAbi from '../../abi/evmTokensAbi.json';
 import { Transaction } from "@mysten/sui/transactions";
 import useSubmitTransaction from "../../api/hooks/useSubmitTransaction";
 
+import { ipAddress, next } from '@vercel/edge'
+import { Ratelimit } from '@upstash/ratelimit'
+import { kv } from '@vercel/kv'
+
 
 const aptosFaucetAddress = '0x275f508689de8756169d1ee02d889c777de1cebda3a7bbcce63ba8a27c563c6f';
```
Comment (l-monninger) [inline src/pages/LandingPage/Index.tsx:25]: Why are these here? This should just be calling the faucet upstream. If these are for testing, leave a comment and also make these reusable.

**Review by l-monninger — CHANGES_REQUESTED** [review]

File: src/pages/LandingPage/Index.tsx (hunk 3 above)
Comment (0xPrimata) [reply src/pages/LandingPage/Index.tsx:25]: these are used for rate limiting

File: api/rate-limit.ts [hunk 4]
```diff
@@ -119,13 +119,18 @@ export default async function handler(request: any, response: any) {
     if (!success) {
       return response.status(429).json({success: false, error: "Rate limited"});
     }
+
+    const HEADERS: Record<string, string | number | boolean> = {
+      Authorization: `Bearer ${process.env.FAUCET_AUTH_TOKEN}`
```
Comment (l-monninger) [inline api/rate-limit.ts:124]: Should be good.

**Review by l-monninger — APPROVED** [review]
